python/ambassador/ir/irerrorresponse.py

```diff
@@ -180,10 +215,19 @@ def _generate_mappers(self) -> Optional[List[Dict[str, Any]]]:
                     continue
 
                 body_format_override["text_format_source"] = ir_text_format_source
+                try:
+                    fmt_file = open(ir_text_format_source["filename"], mode='r')
```

Comment:
This will raise an exception if, somehow, `ir_text_format_source` has no `filename` element...

Reply:
That case is ruled out earlier in the following code block:
```python
 209             if ir_text_format_source is not None:                                                                                                                                                                                      
 210                 # Verify that the text_format_source field is an object with a string filename.                                                                                                                                        
 211                 if not isinstance(ir_text_format_source, dict) or \                                                                                                                                                                    
 212                         not isinstance(ir_text_format_source.get('filename', None), str):                                                                                                                                              
 213                     self.post_error(                                                                                                                                                                                                   
 214                             f"IRErrorResponse: text_format_source field must be an object with a single filename field, found \"{ir_text_format_source}\"")                                                                            
 215                     continue                                                                                                                                                                                                           
 ```
 
 We can assume that that field exists and is a string.

Reply:
Works for me!